src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -42,12 +77,84 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
-                // TODO: Implement the actual export logic here.
+                // Get destination type from secret store.
+                DestinationInfo destinationInfo = await GetDestinationInfo(cancellationToken);
+
+                // Connect to the destination using appropriate client.
+                _exportDestinationClient = _exportDestinationClientFactory.Create(destinationInfo.DestinationType);
+
+                await _exportDestinationClient.ConnectAsync(destinationInfo.DestinationConnectionString, cancellationToken);
+
+                // TODO: For now, always restart from the beginning. We will support resume in another work item.
+                exportJobRecord.Progress = new ExportJobProgress(continuationToken: null, page: 0);
+
+                ExportJobProgress progress = exportJobRecord.Progress;
+
+                // Current page will be used to organize a set of search results into a group so that they can be committed together.
+                uint currentPage = progress.Page;
```

Comment:
nit: currentBatch instead of currentPage? Makes the distinction more clear.

Reply:
I renamed it to currentBatchId. We can't do `++` because it increments by NumberOfPagesPerCommit not by 1.